pyrobosim/pyrobosim/core/robot.py

```diff
@@ -207,11 +217,35 @@ def pick_object(self, obj_query):
             )
             return False
 
+        # If a grasp generator has been specified and no explicit grasp has been provided,
+        # generate grasps here.
+        # TODO: Specify allowed grasp types
+        if grasp_pose is not None:
+            self.latest_grasp = Grasp(origin=grasp_pose)
+        elif self.grasp_generator is not None:
+            cuboid_pose = Pose.from_transform(
+                np.matmul(obj.cuboid_pose.get_transform_matrix(),
+                          obj.pose.get_transform_matrix(),      
+                )
+            )
+            grasps = self.grasp_generator.generate(
+                obj.cuboid_dims, cuboid_pose, self.pose,
+                front_grasps=True, top_grasps=True, side_grasps=False
+            )
+
+            if len(grasps) == 0:
+                warnings.warn(f"Could not generate valid grasps. Cannot pick.")
+                return False
+            else:
+                # TODO: For now, just pick a random grasp.
+                self.latest_grasp = np.random.choice(grasps)
+                print(self.latest_grasp)
```

Comment:
NOTE: In TAMP, grasps would be selected based on some cost function like distance from robot base, or even have some cost incentives to e.g. favor top grasps